src/SqlAsyncEnumerable.cs

```diff
@@ -101,23 +107,22 @@ public ValueTask<bool> MoveNextAsync()
             /// </returns>
             private async Task<bool> GetNextRowAsync()
             {
-                if (this._reader == null)
+                if (this._connection.State != System.Data.ConnectionState.Closed)
```

Comment:
Could you debug and add a breakpoint in the dispose call to figure out where that's coming from? Because the fact that it's disposing the enumerable while something (the function) is still using it is worrisome to me.

Reply:
![image](https://user-images.githubusercontent.com/12754347/190013254-53a0f28c-1d69-4789-883a-34dcc4005c1f.png)

Reply:
GetProductsAsyncEnumerable also invokes the same code but it doesn't call  MoveNextAsync after the dispose, probably depends on how you invoke the enumerator.

Reply:
Did you run "Show External Code"?

https://stackoverflow.com/questions/11629340/what-does-external-code-in-the-call-stack-mean

(I usually prefer to use VS for this reason since it's a lot better about this stuff)

Reply:
![image](https://user-images.githubusercontent.com/12754347/190028852-6a91c13c-0367-4326-8c1e-f825b513473c.png)